### PR TITLE
refactor(source): extend WMTSSource from TMSSource.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
Extend `WMTSSource` from `TMSSource`.

### CODE REVIEW

File: src/Source/WMTSSource.js [hunk 1]
```diff
@@ -78,54 +76,15 @@ class WMTSSource extends Source {
...
-
-        this.format = this.format || 'image/png';
-        this.version = source.version || '1.0.0';
-        this.tileMatrixSet = source.tileMatrixSet || 'WGS84';
-        this.style = source.style || 'normal';
-        this.name = source.name;
         this.url = `${this.url}` +
-            `?LAYER=${this.name}` +
+            `?LAYER=${source.name}` +
             `&FORMAT=${this.format}` +
```
Comment (zarov) [inline src/Source/WMTSSource.js:81]: You forgot to add a default format

File: src/Source/TMSSource.js [hunk 2]
```diff
@@ -65,21 +65,33 @@ class TMSSource extends Source {
...
+                const size = arrayLimits.length;
+                const maxZoom = Number(arrayLimits[size - 1]);
+                const minZoom = maxZoom - size + 1;
+
+                this.zoom = {
+                    min: minZoom,
+                    max: maxZoom,
+                };
+            } else {
+                this.zoom = { min: 2, max: 20 };
```
Comment (zarov) [inline src/Source/TMSSource.js]: Don't forget to update the documentation above.

**Review by zarov — COMMENTED** [review]
Looks good, nothing missing it seems.

File: src/Source/WMTSSource.js (hunk 1 above)
Comment (gchoqueux) [reply src/Source/WMTSSource.js:81]: is already in `TMSSource`

File: src/Source/TMSSource.js (hunk 2 above)
Comment (gchoqueux) [reply src/Source/TMSSource.js]: changed

**Review by zarov — APPROVED** [review]

**Pull request merged**